rpg_game.py

```diff
@@ -161,101 +226,217 @@ def __init__(self):
+            hero_action(int): An interger that corresponds to a choice the user
+            made during the current turn in battle.
+
+        Returns:
+            Bool: True if the hero is still in combat. False if the hero has
+            fled from battle."""
+
+        if hero_action == 1:
+            hero.attack(enemy)
+            enemy.attack(hero)
+            return True
+        elif hero_action == 2:
+            print("")
+            print("That was dumb...?")
+            print("")
+            enemy.attack(hero)
+            return True
+        else:
+            print("Running away... ...")
+            return False
+
+
+    def post_battle_text(self, hero, enemy, shop):
+        """Displays post-battle information to the user. Gives the player the
+        option to go to the store if they are still alive.
+
+        Args:
+            hero: A hero object controlled by the player.
+            enemy: The enemy object the hero is battling against.
+            shop: The shop object, called after the battle to allow the hero
+            to shop.
+
+        Returns:
+            Bool: True if the hero is alive. False if the hero has died."""
+
+        print("The battle has ended...")
+
+        if hero.alive() and not enemy.alive():
             hero.coins += enemy.bounty
             print("You have defeated the {}!".format(enemy.name))
-            return True
+        elif hero.alive() and enemy.alive():
+            print("You are safe, for now.")
         else:
-            print("You have dishonored your ancestors.")
+            print("You have perished.")
             return False
- 
 
+        go_to_store = InputManager.get_yes_or_no("Shall we go to the store then? (y/n)")
+        if go_to_store:
+            shop.welcome()
+            shop.do_shopping(hero)
+        else:
+            print("")
+            print("Fool! Get Ready to face your next opponenet.")
+
+        return True
+
+
+    def battle(self, hero, enemy, shop):
+        """Main battle loop for the Thunderdome class. Handles battle logic.
+
+        Args:
+            hero: A hero object controlled by the player.
+            enemy: The enemy object the hero is battling against.
+            shop: The shop object, called after the battle to allow the hero
+            to shop.
+
+        Returns:
+            Bool: True is the hero is still alive. False if the hero has died."""
 
+        self.print_pre_battle_prompt()
+        self.decide_if_battle(enemy)
+
+        while hero.alive() and enemy.alive():
+            self.print_battle_data(hero, enemy)
+            hero_action = self.decide_hero_action(hero, enemy)
+            still_in_battle = self.resolve_hero_action(hero, enemy, hero_action)
+            if not still_in_battle:
+                break
+
+        hero_alive = self.post_battle_text(hero, enemy, shop)
+        if hero_alive:
+            return True
+        return False
```

Comment:
The bulk of what I did was here, most of it was just splitting up the battle loop so that the logic was a little more clear. Whether or not I was successful is up to you.

This was also where the logic issue I was discussing was. Because the store was called no matter what in the main function, it didn't matter if the player said no to visiting the store.

Reply:
I really like what you did here. The logic is much more clear.